src/ripple/rpc/handlers/WalletPropose.cpp

```diff
@@ -85,25 +85,55 @@ Json::Value walletPropose (Json::Value const& params)
         keyType = keyTypeFromString (
             params[jss::key_type].asString());
 
-        if (keyType == KeyType::invalid)
+        if (!keyType)
             return rpcError(rpcINVALID_PARAMS);
     }
 
-    if (params.isMember (jss::passphrase) ||
-        params.isMember (jss::seed) ||
-        params.isMember (jss::seed_hex))
+    // ripple-lib encodes seed used to generate an Ed25519 wallet in a
+    // non-standard way. While we never encode seeds that way, we try
+    // to detect such keys to avoid user confusion.
     {
```

Comment:
I found myself reading this over and over to see the purpose of the limited scope.  I finally convinced myself that it has no purpose, and was just left over from the removed `if`.

Reply:
I wanted group the ripple-lib specific workaround together in one block.